### ISSUE TITLE
spaghetti needs an api.py module

### ISSUE THREAD

**Issue by sjsrey** [pysal/spaghetti#15]
Similar to what we have done in [libpysal](https://github.com/pysal/libpysal/blob/master/libpysal/api.py).

**Comment by jGaboardi** [comment]
@sjsrey Should this be an exact copy of the `api.py` file in `libpysal`?

**Comment by sjsrey** [comment]
@jGaboardi not an exact copy, but rather use that as a model. Maybe a better example is what we did in [pointpats](https://github.com/sjsrey/pointpats/blob/master/pointpats/api.py)

The idea is to lift functions/classes from deep down in the namespace to the top-level so a user could do something like:

```
import spaghetti.api as spg

ntw = spg.Network('streets.shp')
```
Rather than using the full namespace hierarchy as a developer would:

```
import spaghetti
ntw = spaghetti.network.Network('streets.shp')
```

Having the `api.py` module would enable the "user-mode", which could sit alongside the "developer-mode". Thus we support two different use cases for the code.

**Comment by sjsrey** [comment]
closed with #16
